addons/product/models/product_template.py

```diff
@@ -538,7 +538,29 @@ def copy(self, default=None):
             default = {}
         if 'name' not in default:
             default['name'] = _("%s (copy)", self.name)
-        return super(ProductTemplate, self).copy(default=default)
+
+        res = super(ProductTemplate, self).copy(default=default)
+
+        # Since we don't copy the attribute lines directly, we need to match the extra price
+        # and create the exclusions of the newly created attribute line values.
+        # This matching expects these values to be uniquely determined by the attribute
+        # value. If two lines uses the same attribute value, it will not work properly.
```

Comment:
Honestly, simply having this kind of remark kinda sound like 'this fix does not work'.
There is no 'it will not work properly in this situation', either you make it work, or you specifically catch that situation and skip it, to clearly say that it's not supported and we won't support it, instead of leaving a unknow path, where you don't know what you'll end up with.

Reply:
I might be wrong about this, but I don't think that users should be allowed to associate the same attribute with the same values say: Color W/B, Color W/B and then associate the Black Color with 2 different extra prices. Yet it is possible to achieve this with the current UI. Concerning the extra price match, this is the only unsupported use case (the copied extra price will only be set to the values associated with the last occurrences of W/B).

In such a situation, notice that the mapping attribute value -> Set(attribute value) is not an injective function and hence does not admit an inverse. As such it is impossible to associate deterministically the copied attribute value to their original value.It is still possible of course to associate the pool of extra prices with the copied values to also support this corner case  but the code will then become a gaz factory to support only the case where IMO attribute values are not used properly. I was therefore advised to not support this corner case. But I thought it was therefore important to pint point this.

Would you advise me to remove the two last comment lines or to support the corner case ?

Reply:
PS: Skipping this use case to not setting any extra price, will still set an unexpected value of 0) so IMO its probably as bad as copying the last extra price value. Also, checking that we are in this use case will clearly lower the performances of the operation.

Reply:
> but I don't think that users should be allowed to associate the same attribute with the same values say: Color W/B, Color W/B and then associate the Black Color with 2 different extra prices.

Well, a lot of things can happen, and it's not because it a strange/dumb/... behavior that nobody does it (and there might be a situation where it's meaningful).

IMHO, it shouldn't be impossible to correctly handle this case as well:

```python
        for ptal, copied_ptal in zip(self.attribute_line_ids, res.attribute_line_ids):
            for ptav, copied_ptav in zip(ptal.product_template_value_ids, copied_ptal.product_template_value_ids):
                if not ptav.price_extra:
                    continue
                # Maybe add a security check here, making sure that the ptavs attr & value are identical
                copied_ptav.price_extra = ptav.price_extra
```
would do the trick IMHO (maybe a sorted() call might be useful to enforce the order of the records being given to `zip` but not even sure that's necessary).  Normally, the ordering of PTAL & PTAV when copying a product should stay the same, and relying on it, you should be able to map the copied ptav to the original one.

Reply:
Oooh that is a very interesting trick ! Fair enough, I didn't notice that the order of records was determined by the ORM depending on their attribute values.